drv/fpga-devices/src/ecp5.rs

```diff
@@ -0,0 +1,453 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use super::Fpga;
+use bitfield::bitfield;
+use drv_fpga_api::{DeviceState, FpgaError};
+use ringbuf::*;
+use userlib::*;
+use zerocopy::{AsBytes, FromBytes};
+
+/// ECP5 IDCODE values.
+#[derive(Copy, Clone, Debug, FromPrimitive, PartialEq, AsBytes)]
+#[repr(u8)]
```

Comment:
Why not make this a `#[repr(u32)]` using the values (`0x21111043`, etc) below, so you get `FromPrimitive` / `ToPrimitive` for free?

Reply:
Thanks, this cleans up nicely.